### PR TITLE
Update profile in Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As a user I should be able to edit my profile so I can update my personal information or fill incomplete fields](https://trello.com/c/QgBZ6BVI)

---

#### Description:

* Add permitted parameters to *update profile* action 
* Add request spec test to validate *update profile* functionality

### CODE REVIEW

**Review by Alxzu — APPROVED** [review]

**Review by Alxzu — APPROVED** [review]

**Review by gabrielbursztein2 — COMMENTED** [review]
In my opinion, this endpoint should aim to `/user`, not Auth. That way you'll need to create a new UsersController.

**Comment by afmicc** [comment]
> In my opinion, this endpoint should aim to `/user`, not Auth. That way you'll need to create a new UsersController.

@gabrielbursztein2, I applied your recommendations! Could you check it?

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module Api
+  module V1
+    class UsersController < ApplicationController
+      before_action :authenticate_api_v1_user!
+
+      def update
+        current_api_v1_user.update!(update_params)
+        render json: current_api_v1_user, status: :ok
```
Comment (JuanfraM) [inline app/controllers/api/v1/users_controller.rb]: you should call the view `render :show` for this purpose. 
Remember you can define a view :
views/api/v1/users/show.json.jbuilder

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      end
+
+      private
+
+      def update_params
+        params.permit(:name, :gender)
```
Comment (JuanfraM) [inline app/controllers/api/v1/users_controller.rb]: params for user should be under the user resource as:
`params.require(:user).permit(:name, :gender)`
You can check the docs:
[params](https://api.rubyonrails.org/classes/ActionController/Parameters.html)

File: config/rails_best_practices.yml [hunk 3]
```diff
@@ -23,7 +23,7 @@ ProtectMassAssignmentCheck: { }
 RemoveEmptyHelpersCheck: { }
 #RemoveTabCheck: { }
 RemoveTrailingWhitespaceCheck: { }
-RemoveUnusedMethodsInControllersCheck: { except_methods: ['Api::V1::RegistrationsController#sign_up_params'] }
+RemoveUnusedMethodsInControllersCheck: { except_methods: ['Api::V1::RegistrationsController#sign_up_params', 'Api::V1::RegistrationsController#account_update_params'] }
```
Comment (JuanfraM) [inline config/rails_best_practices.yml]: You're not using `account_update_params ` anymore, so you can delete this exception

**Review by gabrielbursztein2 — APPROVED** [review]
Approved after Juanfra's comments

File: app/views/api/v1/users/_info.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,5 @@
+json.id       user.id
+json.email    user.email
+json.name     user.name
+json.uid      user.uid
```
Comment (JuanfraM) [inline app/views/api/v1/users/_info.json.jbuilder]: `uid` shouldn't be returned. It is used for authentication, it's part of the header response

File: spec/request/api/v1/targets/create_spec.rb [hunk 5]
```diff
@@ -3,11 +3,15 @@
 describe 'Create Targets', type: :request do
   let!(:user) { create(:user, :confirmed) }
   let!(:auth_header) { user.create_new_auth_token }
-  let(:params) { attributes_for(:target) }
+  let(:params) do
+    {
+      target: build(:target).attributes
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/create_spec.rb]: `attributes_for(:target)` should do the trick

File: spec/request/api/v1/targets/create_spec.rb [hunk 6]
```diff
@@ -26,19 +30,19 @@
...
         expect(json_value(body, 'longitude').to_d.round(decimal_scale))
-          .to eq params[:longitude].round(decimal_scale)
+          .to eq params[:target]['longitude'].round(decimal_scale)
       end
     end
 
     context 'when the user is not logged in' do
       it 'is expected an unauthorized response' do
-        post api_v1_targets_path, params: params, as: :json
+        post api_v1_targets_path, params: params
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/create_spec.rb:45]: why did you remove `as: :json`?

File: spec/request/api/v1/targets/index_spec.rb [hunk 7]
```diff
@@ -7,7 +7,7 @@
 
   describe 'GET api/v1/targets' do
     before do
-      get api_v1_targets_path, headers: auth_header, as: :json
+      get api_v1_targets_path, headers: auth_header
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/targets/index_spec.rb:10]: Same here.

File: app/views/api/v1/users/show.json.jbuilder [hunk 8]
```diff
@@ -0,0 +1 @@
+json.partial! 'info', user: user
```
Comment (JuanfraM) [inline app/views/api/v1/users/show.json.jbuilder]: `json.user user, partial: 'info', as: :user`

File: config/locales/en.yml [hunk 9]
```diff
@@ -31,3 +31,6 @@
 
 en:
   hello: "Hello world"
+  api:
+    errors:
+      invalid_model: "Some fields are invalid"
```
Comment (JuanfraM) [inline config/locales/en.yml]: use single quotes

**Review by JuanfraM — APPROVED** [review]
nice work 💪
